deeplake/client/client.py

```diff
@@ -47,18 +47,22 @@ class DeepLakeBackendClient:
     def __init__(self, token: Optional[str] = None):
         self.version = deeplake.__version__
         self.auth_header = None
-        if token is None:
-            self.token = self.get_token()
-        else:
-            self.token = token
+        self.token = token or self.get_token()
         self.auth_header = f"Bearer {self.token}"
 
+        # remove public token, otherwise env var will be ignored
+        # we can remove this after a while
+        orgs = self.get_user_organizations()
+        if orgs == ["public"]:
+            remove_token()
+            self.token = token or self.get_token()
+            self.auth_header = f"Bearer {self.token}"
```

Comment:
We can keep this or the users will have to do `activeloop logout` which is not nice because they did not login.